bin/xml/impl/glassfish/deploy.xml

```diff
@@ -33,28 +31,40 @@
     </presetdef>
 
     <!-- Deployment: DO NOT EDIT -->
-    <target name="-deploy">
-
-        <if>
-            <isset property="archive.file"/>
-            <then>
-                <fileset file="${archive.file}" id="deploy.archive.set"/>
-                <gf.echo message="Deploying:  ${archive.file}"/>
-            </then>
-        </if>
-
-        <ts.glassfish.undeploy>
-            <fileset refid="deploy.archive.set"/>
-        </ts.glassfish.undeploy>
-        
-        <ts.add.sun.xml.to.component.archive> 
-            <fileset refid="deploy.archive.set"/>
-        </ts.add.sun.xml.to.component.archive>
-        
-        <ts.glassfish.deploy>
-            <fileset refid="deploy.archive.set"/>
-        </ts.glassfish.deploy>
-    </target>
+	<!-- Deployment: DO NOT EDIT -->
+	<target name="-deploy">
+
+		<if>
+			<isset property="archive.file"/>
+			<then>
+				<fileset file="${archive.file}" id="deploy.archive.set"/>
+				<gf.echo message="Deploying:  ${archive.file}"/>
+			</then>
+		</if>
+		
+		<ts.glassfish.undeploy>
+			<fileset refid="deploy.archive.set" />
+		</ts.glassfish.undeploy>
+
+		<ts.add.sun.xml.to.component.archive>
+			<fileset refid="deploy.archive.set" />
+		</ts.add.sun.xml.to.component.archive>
+
+		<ts.glassfish.deploy>
+			<fileset refid="deploy.archive.set"/>
+		</ts.glassfish.deploy>
+		
+		<!-- Client stub generation-->
+		<if>
+			<isset property="get.stub.clients"/>
```

Comment:
@gerdogdu should we undeploy the EJB client stubs when the client application is undeployed?